src/components/CredentialCard/ShareCredentialCard.tsx

```diff
@@ -0,0 +1,227 @@
+import { browser } from 'webextension-polyfill-ts';
+import { useCallback, useState, useEffect, useRef } from 'react';
+import { includes, without } from 'lodash-es';
+import cx from 'classnames';
+
+import { Credential } from '../../utilities/credentials/credentials';
+import { usePopupData } from '../../utilities/popups/usePopupData';
+import { Identity } from '../../utilities/identities/types';
+
+import { ShareInput } from '../../channels/shareChannel/types';
+
+import { useScrollIntoView } from './CredentialCard';
+import { Selected } from '../../views/ShareCredential/ShareCredential';
+
+import * as styles from './CredentialCard.module.css';
+
+function useRequiredProperties(credential: Credential, data: ShareInput) {
+  const [requiredProperties, setRequiredProperties] = useState<string[]>([]);
+
+  useEffect(() => {
+    const { cTypes } = data.credentialRequest;
+
+    const cType = cTypes.find(
+      ({ cTypeHash }) => cTypeHash === credential.request.claim.cTypeHash,
+    );
+    setRequiredProperties(cType?.requiredProperties || []);
+  }, [data, credential]);
+
+  return requiredProperties;
+}
+
+interface Props {
+  credential: Credential;
+  identity: Identity;
+  onSelect: (value: Selected) => void;
+  isSelected: boolean;
+}
+
+export function ShareCredentialCard({
+  credential,
+  identity,
+  onSelect,
+  isSelected = false,
+}: Props): JSX.Element {
+  const t = browser.i18n.getMessage;
+
+  const [expanded, setExpanded] = useState(false);
+
+  useEffect(() => {
+    if (isSelected) {
+      setExpanded(true);
+    } else {
+      setExpanded(false);
+    }
+  }, [isSelected]);
+
+  const statuses = {
+    pending: t('component_CredentialCard_pending'),
+    attested: t('component_CredentialCard_attested'),
+    revoked: t('component_CredentialCard_revoked'),
+  };
+
+  const contents = Object.entries(credential.request.claim.contents);
+
+  const handleCollapse = useCallback(() => setExpanded(false), []);
+
+  const data = usePopupData<ShareInput>();
+
+  const [checked, setChecked] = useState<string[]>([]);
```

Comment:
Let’s not touch it now, but when modifying a list of strings, the simplest storage for them is `{}`, or even a `Set`